--- a/Makefile
+++ b/Makefile
@@ -465,8 +465,6 @@
 	src/input/mpegts/iptv/iptv_udp.c \
 	src/input/mpegts/iptv/iptv_rtsp.c \
 	src/input/mpegts/iptv/iptv_rtcp.c \
-	src/input/mpegts/iptv/iptv_pipe.c \
-	src/input/mpegts/iptv/iptv_file.c \
 	src/input/mpegts/iptv/iptv_auto.c
 SRCS-${CONFIG_IPTV} += $(SRCS-IPTV)
 I18N-C += $(SRCS-IPTV)
--- a/src/input/mpegts/iptv/iptv.c
+++ b/src/input/mpegts/iptv/iptv.c
@@ -1292,8 +1292,6 @@
   iptv_http_init();
   iptv_udp_init();
   iptv_rtsp_init();
-  iptv_pipe_init();
-  iptv_file_init();
 #if ENABLE_LIBAV
   iptv_libav_init();
 #endif
--- a/src/input/mpegts/iptv/iptv_private.h
+++ b/src/input/mpegts/iptv/iptv_private.h
@@ -231,8 +231,6 @@
 void iptv_http_init    ( void );
 void iptv_udp_init     ( void );
 void iptv_rtsp_init    ( void );
-void iptv_pipe_init    ( void );
-void iptv_file_init    ( void );
 void iptv_libav_init   ( void );
 
 ssize_t iptv_rtp_read(iptv_mux_t *im, void (*pkt_cb)(iptv_mux_t *im, uint8_t *buf, int len));
